Does not build with GCC 10

Does not build on Ubuntu 18.04 with GCC 10 for me:

```
/usr/bin/g++-10  -DRR_VERSION=\"5.4.0\" -Drr_EXPORTS -I../include -I../third-party/proc-service -I../third-party/brotli/include -I. -D__USE_LARGEFILE64 -pthread -msse2 -D__MMX__ -D__SSE__ -D__SSE2__ -D__STDC_LIM
IT_MACROS -D__STDC_FORMAT_MACROS -std=c++14  -pthread -g   -Wall -Wextra -DDEBUG -UNDEBUG -g3 -Werror -MD -MT CMakeFiles/rr.dir/src/TraceStream.cc.o -MF CMakeFiles/rr.dir/src/TraceStream.cc.o.d -o CMakeFiles/rr.
dir/src/TraceStream.cc.o -c ../src/TraceStream.cc                                
../src/TraceStream.cc: In member function ‘void rr::TraceWriter::write_frame(rr::RecordTask*, const rr::Event&, const rr::Registers*, const rr::ExtraRegisters*)’:
../src/TraceStream.cc:406:29: error: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘class capnp::word’ with no trivial copy-assignment; use value-initialization instead [-Werror=class-memaccess]
  406 |   memset(buf, 0, sizeof(buf));                                                                                                      
      |                             ^                                                                                                     
In file included from /usr/include/capnp/message.h:26,                                                                                      
                 from ../src/TraceStream.cc:5:                                                                                                 
/usr/include/capnp/common.h:329:7: note: ‘class capnp::word’ declared here
  329 | class word { uint64_t content KJ_UNUSED_MEMBER; KJ_DISALLOW_COPY(word); public: word() = default; };
      |       ^~~~
cc1plus: all warnings being treated as errors
```

This is easily fixed:

```
$ git diff
diff --git a/src/TraceStream.cc b/src/TraceStream.cc
index 42109913..ceebc1a6 100644
--- a/src/TraceStream.cc
+++ b/src/TraceStream.cc
@@ -402,8 +402,7 @@ void TraceWriter::write_frame(RecordTask* t, const Event& ev,
   // Use an on-stack first segment that should be adequate for most cases. A
   // simple syscall event takes 320 bytes currently. The default Capnproto
   // implementation does a calloc(8192) for the first segment.
-  word buf[reasonable_frame_message_words];
-  memset(buf, 0, sizeof(buf));
+  word buf[reasonable_frame_message_words] = {};
   MallocMessageBuilder frame_msg(buf);
   trace::Frame::Builder frame = frame_msg.initRoot<trace::Frame>();
 
```

--- a/src/TraceStream.cc
+++ b/src/TraceStream.cc
@@ -401,8 +401,7 @@ void TraceWriter::write_frame(RecordTask* t, const Event& ev,
   // Use an on-stack first segment that should be adequate for most cases. A
   // simple syscall event takes 320 bytes currently. The default Capnproto
   // implementation does a calloc(8192) for the first segment.
-  word buf[reasonable_frame_message_words];
-  memset(buf, 0, sizeof(buf));
+  word buf[reasonable_frame_message_words] = {};
   MallocMessageBuilder frame_msg(buf);
   trace::Frame::Builder frame = frame_msg.initRoot<trace::Frame>();
 

